include/http2.hrl

```diff
@@ -85,8 +85,8 @@
 
 -type stream_id() :: non_neg_integer().
 -record(frame_header, {
-    length      :: non_neg_integer(),
-    type        :: frame_type(),
+    length      :: non_neg_integer() | undefined,
```

Comment:
This can't ever be undefined. Either you're able to parse the bytes into a frame header which includes a real number for length, or you can't. If you can't we're looking at a protocol error